src/client/components/string-filter-menu/string-filter-menu.tsx

```diff
@@ -90,6 +90,24 @@ export class StringFilterMenu extends React.Component<StringFilterMenuProps, Str
       .then(
         (dataset: Dataset) => {
           if (!this.mounted) return;
+          // selectedValues is set before this.mounted flag
+          const { selectedValues } = this.state;
+          if (selectedValues && selectedValues.elements.length > 0) {
+            var selectedElements = selectedValues.elements;
+            var dimensionName = dimension.name;
+            var filtered = dataset.data.filter((d) => {
+              var dimValue = d[dimensionName];
+              return selectedElements.indexOf(dimValue !== null ? String(dimValue) : null) === -1;
+            });
+
+            var selectedDummies: Datum[] = selectedElements.map((v) => {
+              var dummyDatum: Datum = {};
+              dummyDatum[dimensionName] = v;
+              return dummyDatum;
+            });
+
+            dataset.data = selectedDummies.concat(filtered);
```

Comment:
I think this code would be better off to live in `render` (or in a preprocess step)